fission/src/systems/physics/PhysicsSystem.ts

```diff
@@ -658,6 +682,91 @@ class PhysicsSystem extends WorldSystem {
         }
     }
 
+    /**
+     * Creates the Jolt ShapeSettings for a given part using the Part Definition of said part.
+     *
+     * @param   partDefinition  Definition of the part to create.
+     * @returns If successful, the created convex hull shape settings from the given Part Definition.
+     */
+    private CreateConcaveShapeSettingsFromPart(
+        partDefinition: mirabuf.IPartDefinition
+    ): [Jolt.ShapeSettings, Jolt.Vec3, Jolt.Vec3] | undefined | null {
+        const settings = new JOLT.MeshShapeSettings()
+
+        settings.mMaxTrianglesPerLeaf = 8
+
+        settings.mTriangleVertices = new JOLT.VertexList()
+        settings.mIndexedTriangles = new JOLT.IndexedTriangleList()
+        settings.mMaterials = new JOLT.PhysicsMaterialList()
+
+        settings.mMaterials.push_back(new JOLT.PhysicsMaterial())
+
+        const min = new JOLT.Vec3(1000000.0, 1000000.0, 1000000.0)
+        const max = new JOLT.Vec3(-1000000.0, -1000000.0, -1000000.0)
+
+        partDefinition.bodies!.forEach(body => {
+            if (
+                body.triangleMesh &&
+                body.triangleMesh.mesh &&
+                body.triangleMesh.mesh.verts &&
+                body.triangleMesh.mesh.indices
+            ) {
+                const vertArr = body.triangleMesh.mesh.verts
+                for (let i = 0; i < body.triangleMesh.mesh.verts.length; i += 3) {
+                    const vert = MirabufFloatArr_JoltFloat3(vertArr, i)
+                    settings.mTriangleVertices.push_back(vert)
+                    this.UpdateMinMaxBounds(new JOLT.Vec3(vert), min, max)
+                    JOLT.destroy(vert)
+                }
+                const indexArr = body.triangleMesh.mesh.indices
+                for (let i = 0; i < body.triangleMesh.mesh.indices.length; i += 3) {
+                    settings.mIndexedTriangles.push_back(
+                        new JOLT.IndexedTriangle(indexArr.at(i)!, indexArr.at(i + 1)!, indexArr.at(i + 2)!, 0)
+                    )
+                }
+            }
```

Comment:
Utilize the `?` operator to reduce redundancy

```suggestion
            const vertArr = body.triangleMesh?.mesh?.verts
            const indexArr = body.triangleMesh?.mesh?.indices
            if (vertArr && indexArr) {
                for (let i = 0; i < vertArr.length; i += 3) {
                    const vert = MirabufFloatArr_JoltFloat3(vertArr, i)
                    settings.mTriangleVertices.push_back(vert)
                    this.UpdateMinMaxBounds(new JOLT.Vec3(vert), min, max)
                    JOLT.destroy(vert)
                }
                for (let i = 0; i < indexArr.length; i += 3) {
                    settings.mIndexedTriangles.push_back(
                        new JOLT.IndexedTriangle(indexArr.at(i)!, indexArr.at(i + 1)!, indexArr.at(i + 2)!, 0)
                    )
                }
            }
```